### DISCUSSION TITLE
object selection issue in IOS safari browser

### DISCUSSION THREAD

**Discussion by pramod-praella** [fabricjs/fabric.js#10128] [General]
Hello 
I am facing an issue with object selection in iPhone IOS safari browser only.
When I click on the object it's not selecting but if I touch somewhere in Canvas it gets selected.
video:- https://drops.praella.com/v/lwiZZg

@asturur @ShaMan123  and other devs please assist me here to resolve it.

I used version 5.3.0 as this project is old one.

**Comment by asturur** [comment]
That looks like exactly something that zoomed in/out your canvas and you didn't call setCoords on the objects.
Or you moved the object programmatically and didn't call setCoords.

Said so i have to be strict here.
Discussions ARE NOT an help request forum with lesser requirement of quality than the issues tab.
You can't post a video and say 'look it doesn't work help me', how people are supposed to know what are you doing.

You can't even just post a giant code blob and expect people to run it.

**Reply by pramod-praella** [reply]
Thanks @asturur for your response,
You are a great man.
I will take care of your words going forward.
Thanks

**Reply by asturur** [reply]
Was setCoords the issue?

**Reply by pramod-praella** [reply]
No, I already took care of that It's a CSS issue that externally zoomed my canvas.
So, when you raised the concern about Zoom I realised it.
Thanks

**Reply by asturur** [reply]
CSS zoom should be taken in account by the touch events, so that is strange.
Maybe you were zooming only one of the 2 layers of the canvas?

**Reply by pramod-praella** [reply]
because of the below css, the issue is happening to me.


@media (max-width: 991px) {
  .main-canvas {
    zoom: 60%;
  }
  .side-out-menu {
    max-height: 80vh;
  }
}
